Changing get method to return object and setting a default time to expire

felipemorais opened the pull request:
Previously we always need use json.parse after "get" method. Except when we want  check if the script exists in localStorage.
I think that "get" can return a object. To check can use "basket.get(key).exist".

I belive that expiration time is needed for all scripts to internal use, so I put "defaultExpiration" to do this. Default value is 5000, i.e. almost 7 months

Commit:
adding file to travis-ci

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Changing get method to return object
edit: removed exist in lib/basket.js

addyosmani commented:
Thanks! Going to play around with this this evening.

sindresorhus commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+  - sudo node test/node-server.js ../../ 80 &
+  - sleep 5
+script: 
+  - phantomjs test/vendor/run-qunit.js "http://localhost:80/test/index.html"
```
No need for `before_script` and `script`. Just add `connect` as devDep to package.json and the test execution to `scripts.test`.

sindresorhus commented on lib/basket.js:
```diff
@@ -78,42 +79,39 @@
 	};
 
-	var wrapStoreData = function( data, expiration ) {
+	var wrapStoreData = function( obj, data) {
```
space

sindresorhus commented on lib/basket.js:
```diff
@@ -78,42 +79,39 @@
-
-		if ( source && ( source.expire && source.expire - +new Date() > 0 ) ) {
+		
+		if( !source.exist || source.expire - +new Date() < 0 ){
```
space

sindresorhus commented on lib/basket.js:
```diff
@@ -78,42 +79,39 @@
+		
+		if( !source.exist || source.expire - +new Date() < 0 ){
+			saveUrl( obj, callback );
+		}else{
```
space

sindresorhus commented:
Why `exists`? Couldn't you just do `!!basket.get(key)`?

felipemorais commented:
When we use `get`. we can do this:
`basket.get("jquery").stamp` but if jquery doesnt exist `(null).stamp` throws error to the document.
So I did `get` always return an object because `({}).stamp` not throws error.

Since `get` always returns an object, the "!" will fail to test if exists.

Thinking better, false could be a good return. `(false).stamps` not throws error and is simple to check.

peol commented:
I would rather not alter the response, it would make much more sense to a user if it returned null/undefined/empty string than an object that has a `exist` key in it. I'd rather check with `typeof` in that case.

felipemorais commented:
I removed travis. I created a branch to do this. 
`get` now returns false if doesn't exist. It's really better! Thanks!
Spaces are correct now.
Squashed my commits.

sindresorhus commented on lib/basket.js:
```diff
@@ -8,6 +8,7 @@
 	var scriptsExecuted = 0;
 	var globalWaitCount = 0;
 	var waitCallbacks = [];
+	var defaultExpiration = 5000;
```
Question to all. Is 5000 too much as default?

peol replied:
I think it's fine. We could even raise it a bit more if we could find a way so that the developer could easily cache bust. Maybe something like `basket.revision(1);` and whenever he/she needs to refresh the cache, they just bump it to `basket.revision(2);` and basket.js would clear all old stuff and store it using rev 2 instead. That way, we could probably have an expiration that would virtually never expire and leave it up to the developer completely.

But that's a whole other discussion I guess ;)

addyosmani replied:
+1. I think it works  at 5000.

sindresorhus replied:
okidoki

sindresorhus commented:
Thanks :)

Keep em coming!
